model/jpa/src/main/java/org/keycloak/connections/jpa/util/JpaUtils.java

```diff
@@ -56,11 +61,22 @@ public static String getTableNameForNativeQuery(String tableName, EntityManager
         return (schema==null) ? tableName : dialect.openQuote() + schema + dialect.closeQuote() + "." + tableName;
     }
 
+    private static List<ParsedPersistenceXmlDescriptor> transformPersistenceUnits(Collection<PersistenceUnitDescriptor> descriptors) {
+        return descriptors.stream().map(descriptor -> (ParsedPersistenceXmlDescriptor) descriptor).collect(Collectors.toList());
```

Comment:
The Persistence unit descriptor is always `ParsedPersistenceXmlDescriptor` when parsed from the persistence XML -> see https://github.com/hibernate/hibernate-orm/blob/7.0.5/hibernate-core/src/main/java/org/hibernate/jpa/boot/spi/PersistenceXmlParser.java#L181